### ISSUE TITLE
Moving development forward

### ISSUE THREAD

**Issue by seabbs** [epinowcast/coerceDT#3]
We have had a fair few months with no progress on this (for understandable reasons). Shall we setup a meeting with those interested in contributing and hash out some issues that we can PR against to at least get an MVP version that can replace the functionality in `epinowcast`.

**Comment by pearsonca** [comment]
@seabbs could use a second set of eyes on https://github.com/epinowcast/coerceDT/blob/main/R/coerceDT.R - thoughts on the interface offered?

**Comment by pearsonca** [comment]
@seabbs I think coerceDT + tests is basically done, moving on to checkDT.

**Comment by pearsonca** [comment]
Tagging @nikosbosse - I understand you've got an internal helper function like this. If it potentially makes sense to substitute this in, please advise (with the helper function name(s)) and I'll have a look.

**Comment by pearsonca** [comment]
> The version in `scoringutils` is merely this:
> 
> ```
> scoringutils:::ensure_data.table
> function(data) {
>   if (is.data.table(data)) {
>     data <- copy(data)
>   } else {
>     data <- as.data.table(data)
>   }
>   return(data)
> }
> ```
> 
> Maybe there is room for more sophistication :)

Indeed! In terms of just replacing that function, that would literally just be: `ensure_data.table(data)` => `coerceDT(data)` (though now data could be a file path or any other string `fread` just handles).

If you have places internally in scoringutils where you don't need to copy, those would become `coerceDT(data, copy = FALSE)`.

**Comment by pearsonca** [comment]
I did fiddle a bit with S3 methods - found it easier to just handle dispatch based on type in the internal function logic.

**Comment by pearsonca** [comment]
@seabbs per discussion re nomenclature - how about `asDT`?
